pkg/sql/catalog/catprivilege/fix.go

```diff
@@ -138,3 +138,36 @@ func MaybeFixPrivileges(
 	}
 	return changed
 }
+
+// MaybeUpdateGrantOptions iterates over the users of the descriptor and checks
+// if they have the GRANT privilege - if so, then set the user's grant option
+// bits equal to the privilege bits.
+func MaybeUpdateGrantOptions(p *descpb.PrivilegeDescriptor) bool {
+	changed := false
+	if p.Version >= descpb.GrantOptionVersion {
```

Comment:
I think if we're decreasing the version, then we don't need this new version at all since this is the only place where the version is used and checked. I had envisioned adding this new version would make us run the migration only on the old descriptors but if we're decreasing the version regardless then it seems like this new version wouldn't be needed.

Perhaps it makes sense to do this check
```
			if p != nil {
				// If root doesn't have these grant option bits set, then we know
				// that an old node must have clobbered the grant option bits.
				if !p.CheckGrantOptions(security.RootUserName(), privilege.List{privilege.ALL}) {
					p.Version = descpb.Version21_2
				}
			}
```

here instead, if root has grant option bits then we can skip the migration?

Reply:
hm yeah, that seems reasonable. the only point i could see is that setting the version makes things very explicit. but i think ultimately since this migration is not backwards incompatible, you're right we don't need a new version at all.

Reply:
actually, i still think the GrantOptionVersion could be valuable for one more reason: for system tables, admin/root do not have ALL privileges or grant options. We don't want to run `MaybeUpdateGrantOptions` each time we read a system table descriptor, so incrementing the version allows us to avoid that.

Reply:
We could directly use `IsSystemID` and skip if it is right?

Reply:
no, that doesn't work since the admins should still get the SELECT grant option for the system tables; but i think we should be able to use a different check. admins should have the SELECT grant option on every table no matter what, so that should work